src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Formatting/FormattingContext.cs

```diff
@@ -17,9 +17,15 @@ namespace Microsoft.AspNetCore.Razor.LanguageServer.Formatting
 {
     internal class FormattingContext : IDisposable
     {
+        private AdhocWorkspaceFactory _workspaceFactory;
         private Document _csharpWorkspaceDocument;
         private AdhocWorkspace _csharpWorkspace;
 
+        private FormattingContext(AdhocWorkspaceFactory workspaceFactory)
```

Comment:
@davidwengier don't hate me 😄

Reply:
Never!

The workspace stuff here is on the list to be cleaned up, and this probably helps. At the moment the workspaces that are created are never disposed (which is presumably bad) and never re-used, but if its now not doing lots of MEF work thats a step in the right direction.